### ISSUE TITLE
Different solutions to equivalent optimization problems

### ISSUE THREAD

**Issue by ufukguenes** [experimental-design/bofire#228]
For two equivalent optimization problems, two different solutions are provided by the `find_local_max_ipopt` function. Is this expected behavior?
When using the `find_local_max_ipopt` funciton to find the optimal input for the experiments, I can describe the constraints in multiple equivalent ways. Let D(x) be the D-Optimal-Criterium.

```math
 \begin{aligned}
	  \text{minimize} \  D(x) \\
	  s.t. \\
	  
	  0 \leq x_1 \leq 5 \\
	  0 \leq x_2 \leq 15 \\
	  1 \leq x_3 \leq 1 \\
	  \\
	  x_1 + 0.2 * x_2 + 0 * x_3 \leq 5 \\
	  x_1 - x_2 + 0 * x_3 \leq 5 \\
	  \\
	  \Leftrightarrow \\
	  \\
	  \text{minimize} \ D(x)\\
	  s.t. \\
	  
	  0 \leq x_1 \leq 5 \\
	  0 \leq x_2 \leq 15 \\
	  1 \leq x_3 \leq 1 \\
	  \\
	  x_1 + 0.2 * x_2 + 100 * x_3 \leq 105 \\
	  x_1 - x_2 + 100 * x_3 \leq 105\\
  \end{aligned}
```
As $x_3$ is bound to exactly 1 both optimization problems should have the same solution. Writing this in code results in:

```
from bofire.data_models.constraints.api import LinearInequalityConstraint
from bofire.data_models.domain.api import Domain
from bofire.data_models.features.api import (
    ContinuousInput,
    ContinuousOutput,
)
from bofire.strategies.doe.design import find_local_max_ipopt
import numpy as np
import torch
import random

torch.manual_seed(0)

domain = Domain(
        inputs=[
            ContinuousInput(key="x1", bounds=(0, 5)),
            ContinuousInput(key="x2", bounds=(0, 15)),
            ContinuousInput(key="x3", bounds=(1, 1)),
        ],
        outputs=[ContinuousOutput(key="y")],
        constraints=[
            LinearInequalityConstraint(features=["x1", "x2", "x3"], coefficients=[1, 0.2, 0], rhs=5),
            LinearInequalityConstraint(features=["x1", "x2", "x3"], coefficients=[1, -1, 0], rhs=5),
        ],
    )


result = find_local_max_ipopt(
    domain, "linear", n_experiments=3, ipopt_options={"disp": 0}
)
print(result)

```

This returns: 
```
                x1            x2   x3
exp0 -7.099351e-09  1.716973e-09  1.0
exp1  5.000000e+00 -1.515150e-09  1.0
exp2  8.313947e-01  1.500000e+01  1.0
```

When changing the linear inequality constraints to

```
LinearInequalityConstraint(features=["x1", "x2", "x3"], coefficients=[1, 0.2, 100], rhs=105),
LinearInequalityConstraint(features=["x1", "x2", "x3"], coefficients=[1, -1, 100], rhs=105),
```

The result is:
```
                x1            x2   x3
exp0 -8.784069e-09 -6.152546e-09  1.0
exp1  5.000000e+00 -9.999960e-09  1.0
exp2  2.605568e-01  1.500000e+01  1.0
```

Here, x1 in exp2 differs from the value it had in the previous minimization problem. Its value is now around 0.26 where it was at 0.83 before.

**Comment by Osburg** [comment]
Hi @ufukguenes,
thanks for pointing this out! I can reproduce your output. The only difference between both constraint sets you used is the gradient with respect to x3. As far as I know IPOPT takes variables with lower_bound == upper_bound out of the optimization process, so in principle the gradient with respect to `x3` should not be relevant. However, you can see in the implementation of the`LinearConstraint` class that a linear constraint is normalized with the norm of the coefficient vector.

``` 
def __call__(self, experiments: pd.DataFrame) -> pd.Series:
        return (
            experiments[self.features] @ self.coefficients - self.rhs
        ) / np.linalg.norm(self.coefficients)
```

My assumption is that this leads to small numerical differences which lead IPOPT to find different local optima (the D-optimality is the same for both design). Does this sound plausible to you?  Since both optima seem to be equivalent w.r.t D-optimality, the issue you pointed out is not too worrying for me, do you agree?

cheers
Aaron :)

**Comment by ufukguenes** [comment]
Hey @Osburg, 
it seems this is the reason for the different solutions. I also agree that it seems not too worrying. In convex optimization problems like this one, IPOPT should find the same optima and for non-convex problems, we can't predict which local optimum is the global optimum anyway and hence, it shouldn't matter which local optimum is found. Thanks for your answer.

**Comment by Osburg** [comment]
Just as a comment: I think this problem is actually non-convex! It is true that -logdet(X) where X is a matrix is convex, but -logdet(X.T@X) is not convex (this is the reason why IPOPT was chosen as an optimizer and not some other solver specialized for convex problems).
One can also see that the cost function cannot be strictly convex, because from a global minimum we can create many more global minima by interchanging the order of the experiments. So we do not have unique global minima. 
+ we the feasible set is not necessarily convex, if the user does some weird stuff :D
